samples/cpp/videocapture_with_audio.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * This sample demonstrates how to read and display frames from a video file, a camera device or an image sequence.
+ * The sample also demonstrates extracting audio buffers from a video file.
+
+ * Capturing video and audio simultaneously from camera device is not supported by OpenCV.
+*/
+
+
+#include <opencv2/core.hpp>
+#include <opencv2/videoio.hpp>
+#include <opencv2/highgui.hpp>
+#include <iostream>
+
+using namespace cv;
+using namespace std;
+
+string keys =
+"{input i | | Path to input image or video file. Skip this argument to capture frames from a camera.}"
+"{with_audio wt | | set --with_audio if you want to extract audio buffers. If no input video file is provided only audio samples will be extracted}"
+"{help h | | print usage}";
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("This sample demonstrates how to read and display frames from a video file, a camera device or an image sequence \n"
+                 "The sample also demonstrates extracting audio buffers from a video file\n"
+                 "Capturing video and audio simultaneously from camera device is not supported by OpenCV\n"
+                 "Usage:\n --input/-i=<video file, image sequence> Skip this argument to capture frames from a camera\n"
+                 "--with_audio/-wt, set --with_audio if you want to extract audio buffers. If no input video file is provided only audio samples will be extracted\n"
+                 "q,Q,esc -- quit\n"
+                 "You can also pass the path to an image sequence and OpenCV will treat the sequence just like a video\n"
+                 "example: --input=right%02d.jpg\n");
+
+    // Check for 'help' argument to display usage information
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    // Get input file path and check if audio extraction is required
+    string file = parser.get<string>("input");
+    bool isWithAudio = parser.has("with_audio");
+
+    Mat videoFrame;
+    Mat audioFrame;
+    vector<vector<Mat>> audioData;
+    VideoCapture cap;
+
+    // Setup VideoCapture based on user input
+    if (isWithAudio) {
+        vector<int> videoParams{ CAP_PROP_AUDIO_STREAM, 0, CAP_PROP_VIDEO_STREAM, 0, CAP_PROP_AUDIO_DATA_DEPTH, CV_16S };
+        if (file.empty()) {
+            // No input file, extract audio from default device
+            cout << "Extracting audio samples from default device is only supported via MSMF backend" << endl;
+            cap.open(0, CAP_MSMF, videoParams);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open camera and microphone" << endl;
+                return -1;
+            }
+        } else {
+            // Open input file for both video and audio
+            cap.open(file, CAP_ANY, videoParams);
```

Comment:
The sample doesn't work with audio files with GStreamer backend (or any backend other than MSMF).

Reply:
@mshabunin how are you testing?

Reply:
Ubuntu 22.04 with base/good/bad/ugly plugins GStreamer plugins installed.
```
./bin/example_cpp_videocapture_with_audio \
    -i=../opencv_extra/testdata/highgui/audio/test_audio.wav \
    --with_audio
```

Reply:
@mshabunin The sample is supposed to work with video which might have an audio. Please try 

> /opencv_extra/testdata/highgui/videoVID00003-20100701-2204.avi